bangc-ops/mlu_op.h

```diff
@@ -2550,6 +2550,156 @@ mluOpSqrtBackward(mluOpHandle_t handle,
+ * \b num_points_per_voxel is number of point per voxel, \b voxel_num is
+ * number of voxels.
+ *
+ * @param[in] handle
+ * Handle to an MLUOP context that is used to manage MLU devices and
+ * queues in the voxelization operation. For detailed information, see
+ * ::mluOpHandle_t.
+ * @param[in] points_desc
+ * The descriptor of the tensors. For detailed information, see
+ * ::mluOpTensorDescriptor_t.
+ * @param[in] points
+ * Pointer to the MLU memory that stores the input tensor.
+ * @param[in] voxel_size_desc
+ * The descriptor of the tensors. For detailed information, see
+ * ::mluOpTensorDescriptor_t.
+ * @param[in] voxel_size
+ * Pointer to the MLU memory that stores the input tensor.
+ * @param[in] coors_range_desc
+ * The descriptor of the tensors. For detailed information, see
+ * ::mluOpTensorDescriptor_t.
+ * @param[in] coors_range
+ * Pointer to the MLU memory that stores the input tensor.
+ * @param[in] max_points
+ * An integer value which is the maximum points contained in a voxel.
+ * @param[in] max_voxels
+ * An integer value which is the maximum voxels this function create.
+ * @param[in] NDim
+ * An integer value which is the second dimension of coors.
+ * @param[in] deterministic
+ * A bool value which is whether to invoke the non-deterministic
+ * version of hard-voxelization implementations. Currently,
+ * Non-deterministic mode not supported.
+ * @param[in] workspace
+ * Pointer to the MLU memory that stores the extra workspace.
+ * @param[in] workspace_size
+ * The size of extra space.
+ * @param[in] voxels_desc
+ * The descriptor of the tensors. For detailed information, see
+ * ::mluOpTensorDescriptor_t.
+ * @param[in] voxels
+ * Pointer to the MLU memory that stores the input tensor.
+ * @param[in] coors_desc
+ * The descriptor of the tensors. For detailed information, see
+ * ::mluOpTensorDescriptor_t.
+ * @param[in] coors
+ * Pointer to the MLU memory that stores the input tensor.
+ * @param[in] num_points_per_voxel_desc
+ * The descriptor of the tensors. For detailed information, see
+ * ::mluOpTensorDescriptor_t.
+ * @param[in] num_points_per_voxel
+ * Pointer to the MLU memory that stores the input tensor.
+ * @param[in] voxel_num_desc
+ * The descriptor of the tensors. For detailed information, see
+ * ::mluOpTensorDescriptor_t.
+ * @param[in] voxel_num
+ * Pointer to the MLU memory that stores the input tensor.
+ *
+ * @par Return
+ * - ::MLUOP_STATUS_SUCCESS, ::MLUOP_STATUS_BAD_PARAM, ::MLUOP_STATUS_NOT_SUPPORTED.
+ *
+ * @par Data Type
+ * - The supported data types of input and output tensors are as follows:
+ *   - points, voxel_size, coors_range, voxels: float.
+ *   - coors, num_points_per_voxel, voxel_num: int.
+ *
+ * @par Scale Limitation
+ * - max_points and max_voxels must be greater than or equal to 0.
+ * - NDim must be equal to 3, which means 3-D.
+ * - deterministic must be True. Currently, Non-deterministic mode not supported.
```

Comment:
The value of the deterministic mode must be True. Currently, the non-deterministic mode is not supported.